modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/CopyDataDefinitionMVCActionCommand.java

```diff
@@ -101,104 +92,33 @@ protected void doProcessAction(
 			).build();
 
 		DataDefinition dataDefinition =
-			dataDefinitionResource.getDataDefinition(ddmStructureId);
+			dataDefinitionResource.postDataDefinitionCopy(ddmStructureId);
 
-		_uniquifyDataDefinitionFields(dataDefinition);
-
-		dataDefinition.setDataDefinitionKey(StringPool.BLANK);
 		dataDefinition.setDescription(
 			LocalizedValueUtil.toStringObjectMap(descriptionMap));
 		dataDefinition.setName(LocalizedValueUtil.toStringObjectMap(nameMap));
 
-		dataDefinition =
-			dataDefinitionResource.postSiteDataDefinitionByContentType(
-				themeDisplay.getScopeGroupId(), "journal", dataDefinition);
+		dataDefinitionResource.putDataDefinition(
+			dataDefinition.getId(), dataDefinition);
 
 		boolean copyTemplates = ParamUtil.getBoolean(
 			actionRequest, "copyTemplates");
 
 		if (copyTemplates) {
-			DDMStructure ddmStructure = _ddmStructureService.getStructure(
-				themeDisplay.getScopeGroupId(),
-				_portal.getClassNameId(JournalArticle.class),
-				dataDefinition.getDataDefinitionKey());
-
 			ServiceContext serviceContext = ServiceContextFactory.getInstance(
 				DDMStructure.class.getName(), actionRequest);
 
 			_ddmTemplateService.copyTemplates(
 				_portal.getClassNameId(DDMStructure.class), ddmStructureId,
 				_portal.getClassNameId(JournalArticle.class),
-				ddmStructure.getStructureId(),
+				dataDefinition.getId(),
```

Comment:
@devpedrotavares why did you have to change this?

Reply:
I changed it because I thought that it is was not needed to get the ddmStructure. Also, this is how it's currently being done in for document-library. Is it best to just leave it as it was?

Reply:
Got it.. thanks for explaining